### PR TITLE
Convert unit tests to snapshot tests

### LINKED ISSUE #95: Snapshot testing
We can (somewhat) simplify all our testing by switching to `insta` and capturing snapshots: https://insta.rs/docs/cli/

This is how rust tests individual rules: https://github.com/astral-sh/ruff/blob/main/CONTRIBUTING.md#rule-testing-fixtures-and-snapshots

Related: #29

### PR DESCRIPTION
**Pull request by ZedThree**
Closes #95 

Builds on #117

### CODE REVIEW

**Comment by ZedThree** [comment]
The `SuperfluousImplicitNone` test might have issues depending on how you run it, as the diff output may or may not have colours in it. I didn't see an easy way to disable the colour for the tests.

File: fortitude/src/lib.rs [hunk 1]
```diff
@@ -263,6 +277,7 @@ fn format_violation(
         let fix_title = Level::Note.title(fix_msg);
         writeln!(f, "\n{}", renderer.render(fix_title))?;
 
+        // FIXME: turn off colour in tests?
```
Comment (LiamPattinson) [inline fortitude/src/lib.rs:280]: I think this FIXME has already been fixed

File: README.dev.md [hunk 2]
```diff
@@ -54,6 +51,42 @@ the rule is intended to fix. Words such as 'forbid' should be omitted. For examp
 name for the rule that warns of overly long lines is `LineTooLong`, and not something
 like `AvoidLineTooLong` or `KeepLinesShort`.
 
+#### Rule testing: fixtures and snapshots
+
+To test rules, Fortitude uses snapshots of Fortitude's output for a given file (fixture). Generally, there
+will be one file per rule (e.g., `E402.py`), and each file will contain all necessary examples of
```
Comment (LiamPattinson) [inline README.dev.md:57]: References a `.py` file, but should be a `.f90` file

File: README.dev.md [hunk 3]
```diff
@@ -54,6 +51,42 @@ the rule is intended to fix. Words such as 'forbid' should be omitted. For examp
...
+both violations and non-violations. `cargo insta review` will generate a snapshot file containing
+Fortitude's output for each fixture, which you can then commit alongside your changes.
+
+Once you've completed the code for the rule itself, you can define tests with the following steps:
+
+1. Add a Python file to `fortitude/resources/test/fixtures/[category]` that contains the code you
```
Comment (LiamPattinson) [inline README.dev.md:63]: Similarly, should be a Fortran file. There are a few other Python references in this doc.

**Review by LiamPattinson — CHANGES_REQUESTED** [review]
Nice one! I found a few very minor issues, but overall it's looking much cleaner than before.

**Comment by LiamPattinson** [comment]
Since the diff output is more of a temporary placeholder at this stage, I'd be in favour of disabling it for now until we can write up a more robust method.

**Comment by ZedThree** [comment]
I've just deleted the diff output for the time being, with the intention we'll restore it later

**Review by LiamPattinson — APPROVED** [review]

**Comment by ZedThree** [comment]
Oh, the macos-12 runner is deprecated, so that won't work

**Pull request merged**